mwdb/resources/object.py

```diff
@@ -339,3 +339,91 @@ def get(self, type):
 
         schema = ObjectCountResponseSchema()
         return schema.dump({"count": result})
+
+
+class ObjectFavoriteResource(Resource):
+    @requires_authorization
+    def post(self, identifier):
+        """
+        ---
+        summary: Add favorite object
+        description: |
+            Add an existing object as a favorite for the user.
+        security:
+            - bearerAuth: []
+        tags:
+            - object
+        parameters:
+            - in: path
+              name: identifier
+              schema:
+                type: string
+              description: Object identifier
+        responses:
+            200:
+                description: Add favorite object
+            400:
+                description: When request body is invalid
+            404:
+                description: When object doesn't exist.
+            409:
+                description: When object is already added as favorite.
+        """
+
+        user = g.auth_user
+
+        favorite_object = Object.access(identifier)
+
+        if favorite_object is None:
+            raise NotFound("Object not found")
+
+        if favorite_object.favorite:
+            raise Conflict("Object is already added as a favorite")
```

Comment:
I think there is nothing wrong with making this idempotent. User wants to mark the object to be favorite and it's done even if the object was already favorite. We should just return here with some `logger.info`.

The same applies to `DELETE`. More information about REST philosophy can be found in this SO answer: https://stackoverflow.com/questions/6439416/deleting-a-resource-using-http-delete

Reply:
If you apply this suggestion, remember to remove the 409 response from the docs.